tornadox/solvers/ek1.py

```diff
@@ -0,0 +1,192 @@
+"""EK1."""
+
+from collections import namedtuple
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+
+from tornadox.blocks import stepsizes
+from tornadox.blocks.inits import autodiff_first_order
+from tornadox.blocks.sde import ibm
+from tornadox.blocks.step_impl import ek1_projmatfree_d_nu_first_order
+
+
+def ek1_terminal_value(*, ode_dimension, num_derivatives=5):
+    """The traditional EK1, but only for computing the terminal value.
+
+    Uses adaptive steps and PI control.
+    Uses time-varying, scalar-valued diffusion.
+    Uses Taylor-mode initialisation for num_derivatives >= 5,
+    and forward-mode initialisation otherweise.
+
+    Uses a dimension-derivative ordering internally,
+    and does not use projection matrices.
+    """
+
+    EK1State = namedtuple("EK1State", ("u", "dt_proposed", "error_norm", "stats"))
```

Comment:
What do you think about making this a 
```python
from typing import NamedTuple

class EK1State(NamedTuple):
    u
    dt_proposed
    error_norm
    stats
```
maybe even outisde of this function? To me that syntax reads much nicer :D

Reply:
This is possible. I tend to not use it if I dont want to add types, because it looks a bit naked. But this is personal taste, and if you prefer it then fine :)
(The in-line is also a single line, not 5+2 for spacing, but that is really minor :shrug:)

I would like to keep it inside the function because I don't want this data structure to be "too public", if that makes sense. I already feel bad for using a class here :D

Reply:
Hmm I see your points. Feel free to decide either way :)

Reply:
Let's keep it for now and iterate if necessary :)